snuba/datasets/storages/generic_metrics.py

```diff
@@ -181,3 +192,49 @@ def produce_policy_creator() -> DeadLetterQueuePolicy:
         pre_filter=KafkaHeaderSelectFilter("metric_type", InputType.DISTRIBUTION.value),
     ),
 )
+
+
+file = open(
+    "./snuba/datasets/configuration/generic_metrics/storage_distributions_raw.yaml"
+)
+conf_yml = safe_load(file)
+assert isinstance(conf_yml, dict)
+conf = dataclass_from_dict(StorageConfig, conf_yml)
+assert isinstance(conf, StorageConfig)
+
+
+distributions_bucket_storage = WritableTableStorage(
```

Comment:
I see that you create an intermediate dataclass representation for your config and then, you create the data model object from that.
What is the reason for these two to be separate ? IT seems the config object do not do anything and do not add any value.

If they provided at least config validation, they would be more useful.
On the other hand having a data model class and config class for each data model object seems very verbose (every time we want to add one). If rely on functions to build the data model from YAML directly you can skip a lot of these steps, but you would need to figure out how to do validation.

Reply:
The only value the intermediate representation was adding was having everything be typed instead of doing something like `conf["schema"]["columns"]` but it got complicated and I just got kinda carried away. I'm going to use JSONSchema to validate and then just use the dictionaries untyped as is.

Reply:
The other advantage of the intermediate interpretation is for lazy loading. We can load the config files on import into the intermediate classes, and only build the actual object when it's requested. That isn't something I'm sure we need though.

Reply:
You can automatically create TypedDict from json schema. No need to create a class 
https://pypi.org/project/jsonschema-typed-v2/

We tried to use that to generate a TypedDict representation of the Events automatically from the jsonschema produced by relay 
https://github.com/getsentry/sentry-data-schemas/tree/main/py